src/main/java/echo/main/Echo.java

```diff
@@ -0,0 +1,233 @@
+        String eventDescription = eventArray[0];
+        String eventStartTime = eventArray[1];
+        String eventEndTime = eventArray[2];
+        Events eventTask = new Events(eventDescription, eventStartTime, eventEndTime);
+        taskList.addTask(eventTask);
+        return ui.printAddTaskMessage(eventTask, taskList);
+    }
+
+    /**
+     * Handles the command when user types in "delete"
+     * Remove the task at that index provided from taskList
+     *
+     * @param index the index of the task that is required to be deleted in String
+     * @return a string of message after deleting task
+     */
+    public String handleDeleteCommand(String index) {
+        Task deletedTask = taskList.getTaskAndDelete(index);
+        return ui.printDeleteMessage(deletedTask, taskList);
+    }
+
+    /**
+     * Handles the command when user types in "find"
+     *
+     * @param keyword keyword contained in the task description
+     * @return a string of list of all the task with the keyword
+     */
+    public String handleFindCommand(String keyword) {
+        ArrayList<Task> arrayList = taskList.findTask(keyword);
+        return ui.printFoundTask(arrayList);
+    }
+
+    public String handleByeCommand() {
+        try {
+            this.storage.saveTaskList(this.taskList);
+            return ui.bye();
+        } catch (IOException e) {
+            return ui.printErrorMessage(e);
+        }
+    }
+
+    /**
+     * Determines which command the user is giving and calls the respective functions to handle the command
+     *
+     * @param input command and description entered by users e.g. deadline work /by 11-12-2024 2345
+     * @return string of output from chatbot
+     * @throws EchoException if the command in the input is not valid
+     */
+    public String executeInput(String input) throws EchoException {
+        try {
+            String[] replyArray = parser.parseInput(input);
+            String command = replyArray[0];
+            String description = (replyArray.length > 1) ? replyArray[1] : "";
+
+            switch (command) {
+            case "list":
+                return handleListCommand();
+            case "mark":
+                return handleMarkCommand(description);
+            case "unmark":
+                return handleUnmarkCommand(description);
+            case "todo":
+                return handleToDoCommand(description);
+            case "deadline":
+                return handleDeadlineCommand(description);
+            case "event":
+                return handleEventCommand(description);
+            case "delete":
+                return handleDeleteCommand(description);
+            case "find":
+                return handleFindCommand(description);
+            case "bye":
+                return handleByeCommand();
+            default:
+                throw new EchoException("Sorry! I don't get what you mean. Try again with the list of commands above.");
+            }
```

Comment:
Nice work on using separate methods to extract out the logic used for each command. This section of code is very easy to read! 🤩